moped-database/migrations/default/1761074353404_ecapris_funding/up.sql

```diff
@@ -33,9 +48,137 @@ COMMENT ON COLUMN moped_proj_funding.funding_description IS 'A description of th
+COMMENT ON COLUMN moped_proj_funding.unit_long_name IS 'The long name of the unit associated with this funding record from eCAPRIS';
+
+-- Create ecapris_subproject_funding table with column comments
+CREATE TABLE public.ecapris_subproject_funding (
+    id SERIAL PRIMARY KEY,
+    ecapris_subproject_id TEXT NOT NULL,
+    fao_id INTEGER NOT NULL UNIQUE,
+    fdu TEXT NOT NULL,
+    app INT4 NOT NULL,
+    unit_long_name TEXT NOT NULL,
+    subprogram TEXT DEFAULT NULL,
+    program TEXT DEFAULT NULL,
+    created_at TIMESTAMP WITH TIME ZONE DEFAULT NOW() NOT NULL,
+    updated_at TIMESTAMP WITH TIME ZONE DEFAULT NOW() NOT NULL,
+    created_by_user_id INTEGER REFERENCES moped_users (user_id) ON DELETE RESTRICT ON UPDATE CASCADE,
+    updated_by_user_id INTEGER REFERENCES moped_users (user_id) ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+-- Index fdu since we'll be querying by it to avoid duplicates in the combined view (using NOT EXISTS)
+CREATE INDEX idx_ecapris_subproject_funding_fdu
+ON ecapris_subproject_funding (fdu);
+
+-- Index ecapris_subproject_id for our future GraphQL query for the project funding table
+CREATE INDEX idx_ecapris_subproject_funding_subproject_id
+ON ecapris_subproject_funding (ecapris_subproject_id);
+
+COMMENT ON TABLE public.ecapris_subproject_funding IS 'Stores eCAPRIS subproject fund records synced from the FSD Data Warehouse to supplement the moped_proj_funding table records.';
+COMMENT ON COLUMN public.ecapris_subproject_funding.id IS 'Primary key for the table';
+COMMENT ON COLUMN public.ecapris_subproject_funding.ecapris_subproject_id IS 'eCapris subproject ID number';
+COMMENT ON COLUMN public.ecapris_subproject_funding.fao_id IS 'Unique ID for the FDU (Fund-Dept-Unit) from eCAPRIS';
+COMMENT ON COLUMN public.ecapris_subproject_funding.fdu IS 'The FDU (Fund-Dept-Unit) code associated with this funding record from eCAPRIS';
+COMMENT ON COLUMN public.ecapris_subproject_funding.app IS 'The appropriated amount associated with this funding record from eCAPRIS';
+COMMENT ON COLUMN public.ecapris_subproject_funding.unit_long_name IS 'The long name of the unit associated with this funding record from eCAPRIS';
+COMMENT ON COLUMN public.ecapris_subproject_statuses.created_at IS 'Timestamp when the record was created';
+COMMENT ON COLUMN public.ecapris_subproject_statuses.updated_at IS 'Timestamp when the record was last updated';
+COMMENT ON COLUMN public.ecapris_subproject_statuses.created_by_user_id IS 'ID of the user who created the record';
+COMMENT ON COLUMN public.ecapris_subproject_statuses.updated_by_user_id IS 'ID of the user who updated the record';
+
+-- Create a combined_project_funding_view for the project funding UI to consume. This view combines
+-- both moped_proj_funding and ecapris_subproject_funding data and removes duplicates based on FDU.
+CREATE OR REPLACE VIEW combined_project_funding_view AS
```

Comment:
This view follows the same pattern that we used for notes to bring notes together in the UI.